app/bundles/CoreBundle/Doctrine/Type/ArrayType.php

```diff
@@ -39,6 +39,10 @@ function (&$entry) {
 
         $serialized = serialize($value);
 
+        if(strpos($serialized, chr(0)) !== false) {
```

Comment:
@alanhartless, @escopecz, @dbhurley Instead of just checking for null bytes and then throwing an exception, would it be acceptable to simply remove them and not throw the exception?

```php
$serialized = serialize($value);

if (strpos($serialized, chr(0) !== false) {
    $serialized = str_replace(chr(0), '', $serialized);
}

return $serialized;
```